dockerproxy/main.go

```diff
@@ -470,6 +472,35 @@ func authorizeRequest(appName, authToken string) bool {
 func extendDeadline() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		log.Infof("extendDeadline called with user agent: %s", r.UserAgent())
+
+		di, err := disk.GetInfo("/data")
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			log.Errorf("failed to check /data: %s", err)
+			return
+		}
+
+		// prune only if the storage space is too low.
+		err = newInsufficientStorageError(di)
+		if err != nil {
+			client, err := client.NewEnvClient()
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				log.Errorf("failed to create a Docker client: %s", err)
+				return
+			}
+
+			prune(context.Background(), client)
+		}
+
+		// return error if pruning is not enough.
+		err = newInsufficientStorageError(di)
```

Comment:
You probably need to fetch the disk info again before this call.

Reply:
Ah right. Will do.

Reply:
@jeromegn Updated in the last revision.